### PR TITLE
Feature/flask

### PR DESCRIPTION
**Pull request by garyluu**
Creates a validation server to test tool registry services against the ga4gh-discovery yaml

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+FROM ubuntu:latest
+MAINTAINER Gary Luu "gary.luu@oicr.on.ca"
+RUN apt-get update -y
+# install node.js and npm
+RUN apt-get -qq update && apt-get install -y nodejs && apt-get install -y npm && ln -s /usr/bin/nodejs /usr/bin/node
+# install git
+RUN apt-get install git -y
+# install dredd 
+RUN npm install -g dredd
```
Comment (denis-yuen) [inline Dockerfile]: version pin or shrinkwrap this

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# test versions 
+RUN node -v && \
+    npm -v && \
+    dredd --version
+RUN apt-get install -y python-pip python-dev build-essential
+ADD https://api.github.com/repos/ga4gh/tool-registry-validator/compare/develop...HEAD /dev/null
```
Comment (denis-yuen) [inline Dockerfile]: What is this for? (add a comment)

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -o errexit
+set -o pipefail
+set -o nounset
+set -o xtrace
+
+curl -O https://raw.githubusercontent.com/ga4gh/tool-registry-schemas/feature/trsv_changes/src/main/resources/swagger/ga4gh-tool-discovery.yaml
```
Comment (denis-yuen) [inline build.sh]: Just grab this file in `validator/validator.py` to avoid needing this script

File: validator/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,12 @@
+FROM ubuntu:latest
+MAINTAINER Gary Luu "gary.luu@oicr.on.ca"
+RUN apt-get update -y
+RUN apt-get install -y python-pip python-dev build-essential
+RUN ls
+RUN python ../setup.py install
+COPY . /app
+WORKDIR /app
+RUN pip install -r requirements.txt
+CMD ["wget", "https://raw.githubusercontent.com/ga4gh/tool-registry-schemas/feature/trsv_changes/src/main/resources/swagger/ga4gh-tool-discovery.yaml"]
```
Comment (denis-yuen) [inline validator/Dockerfile]: Not sure what's going on here, is this a duplicate or a WIP version of the root Dockerfile?
In any case, only one of these CMDs will apply, the last one it seems https://docs.docker.com/engine/reference/builder/#cmd

File: validator/badge.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+BADGE_URL = 'https://img.shields.io/badge/{0}-{1}-{2}.svg?style=flat'
```
Comment (denis-yuen) [inline validator/badge.py:1]: :+1:

File: validator/compareSwaggerYaml.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+import warnings as warnings
+import os
+import sys
+
+import constants
+
+global_issues = []
+
+
+def compare_definitions(yaml1, yaml2):
```
Comment (denis-yuen) [inline validator/compareSwaggerYaml.py:8]: Remember to add documentation for this file, not a Python expert, but a full example here https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

File: validator/createProcessedYAML.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def append_x_example(paths):
+    for path_key in paths:
+        path_value = paths.get(path_key)
+        parameters = path_value.get('get').get('parameters')
+        if parameters is not None:
+            for parameter in parameters:
+                if parameter.get('in') == 'path':
+                    parameter['x-example'] = parameter.get('name') + '_placeholder'
```
Comment (denis-yuen) [inline validator/createProcessedYAML.py]: :+1:

File: validator/createRelaxedYAML.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    # Removing external docs because it is not used and also because there's a warning for some reason
+    yaml_file.pop('externalDocs', None)
+    remove_headers(yaml_file.get('paths'))
+    file_directory = os.path.dirname(__file__)
+    relaxed_swagger = os.path.join(file_directory, constants.RELAXED_SWAGGER)
+    with open(relaxed_swagger, 'w') as warning_yaml_file:
+        yaml.dump(yaml_file, warning_yaml_file, allow_unicode=True)
+
+
+# Some properties are allowed to be null
```
Comment (denis-yuen) [inline validator/createRelaxedYAML.py]: Which ones actually?

**Review by denis-yuen — COMMENTED** [review]
Thanks for the update, please check on / merge #1  to avoid conflicts

File: validator/Dockerfile (hunk 4 above)
Comment (garyluu) [reply validator/Dockerfile]: left over from experiment

**Comment by garyluu** [comment]
Probably ready for re-review

**Comment by denis-yuen** [comment]
@garyluu, have a merge conflict as discussed

**Comment by garyluu** [comment]
Ready for re-review

**Review by denis-yuen — APPROVED** [review]

File: validator/constants.py [hunk 9]
```diff
@@ -3,5 +3,5 @@
 PREPROCESSED_SWAGGER = 'ga4gh-tool-discovery.yaml'
 EXPECTED_PASSING_TESTS = 9
 GITHUB_BASEURL = 'https://raw.githubusercontent.com/ga4gh/tool-registry-schemas'
-GITHUB_BRANCH = 'feature/trsv_changes'
+GITHUB_BRANCH = 'develop'
```
Comment (denis-yuen) [inline validator/constants.py]: probably want to pick a release rather than float here

**Review by denis-yuen — APPROVED** [review]
FYI, I only checked the last three commits. 
Not sure why this wasn't merged before

**Comment by garyluu** [comment]
I think what happened was we were waiting for other people to take a look at it too so I held off on merging.  But then we all forgot about it.

**Pull request merged**